cmd/root_test.go

```diff
@@ -455,3 +456,108 @@ func TestArgsParsing_GCSAuthFlagsThrowsError(t *testing.T) {
 		})
 	}
 }
+
+func TestArgsParsing_GCSConnectionFlags(t *testing.T) {
+	tests := []struct {
+		name           string
+		args           []string
+		expectedConfig *cfg.Config
+	}{
+		{
+			name: "Test gcs connection flags.",
+			args: []string{"gcsfuse", "--billing-project=abc", "--client-protocol=http2", "--custom-endpoint=www.abc.com", "--experimental-enable-json-read", "--experimental-grpc-conn-pool-size=20", "--http-client-timeout=20s", "--limit-bytes-per-sec=30", "--limit-ops-per-sec=10", "--max-conns-per-host=1000", "--max-idle-conns-per-host=20", "--sequential-read-size-mb=70", "abc", "pqr"},
+			expectedConfig: &cfg.Config{
+				GcsConnection: cfg.GcsConnectionConfig{
+					BillingProject:             "abc",
+					ClientProtocol:             "http2",
+					CustomEndpoint:             "www.abc.com",
+					ExperimentalEnableJsonRead: true,
+					GrpcConnPoolSize:           20,
+					HttpClientTimeout:          20 * time.Second,
+					LimitBytesPerSec:           30,
+					LimitOpsPerSec:             10,
+					MaxConnsPerHost:            1000,
+					MaxIdleConnsPerHost:        20,
+					SequentialReadSizeMb:       70,
+				},
+			},
+		},
+		{
+			name: "Test default gcs connection flags.",
+			args: []string{"gcsfuse", "abc", "pqr"},
+			expectedConfig: &cfg.Config{
+				GcsConnection: cfg.GcsConnectionConfig{
+					BillingProject:             "",
+					ClientProtocol:             "http1",
+					CustomEndpoint:             "",
+					ExperimentalEnableJsonRead: false,
+					GrpcConnPoolSize:           1,
+					HttpClientTimeout:          0,
+					LimitBytesPerSec:           -1,
+					LimitOpsPerSec:             -1,
+					MaxConnsPerHost:            0,
+					MaxIdleConnsPerHost:        100,
+					SequentialReadSizeMb:       200,
+				},
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			var gotConfig *cfg.Config
+			cmd, err := NewRootCmd(func(cfg *cfg.Config, _ string, _ string) error {
```

Comment:
Nit:
```suggestion
			cmd, err := NewRootCmd(func(cfg *cfg.Config, _, _ string) error {
```

Reply:
Oops, sorry.. forgot to do this on this PR after rebasing.